### PR TITLE
Add missing backup recovery functionality

### LINKED ISSUE #15886: [BUG] session.xml backup paths are stored as absolute paths but the backups themselves are stored relative to npp's folder which breaks session.xml if npp's folder were renamed or moved
### Is there an existing issue for this?

- [x] I have searched the existing issues
Related: https://github.com/notepad-plus-plus/notepad-plus-plus/issues/5098

### Description of the Issue

So, session.xml backup paths are stored as absolute paths but the backups themselves are stored relative to npp's folder which breaks session.xml if npp being launched after its folder was renamed or moved. This just happens to me between windows reinstallations as I use it in portable mode.
It can also break absolutely easy if npp is used from flash drive and windows just assign another letter to the drive

### Steps To Reproduce

1. unpack portable npp
2. start npp
3. make new file
4. edit it
5. close npp
6. check session.xml that it have absolute path to new file's backup
7. rename or move npp's folder
8. start npp
9. you lost your unsaved edited new file

### Current Behavior

as described in steps 7~9

### Expected Behavior

as backups stored relative to npp's executable folder, session.xml should also have paths that are relative to npp's executable.

Current: `backupFilePath="D:\Users\Rustam\Desktop\0Notepad++\backup\new 5@2024-11-30_165530"`
Expected: `backupFilePath="backup\new 5@2024-11-30_165530"`

### Debug Information

```shell
Notepad++ v8.7.1   (64-bit)
Build time : Oct 31 2024 - 00:48:56
Path : D:\Users\kww\Desktop\0Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : ON
Cloud Config : OFF
Periodic Backup : ON
OS Name : Windows 10 Enterprise LTSC 2021 (64-bit)
OS Version : 21H2
OS Build : 19044.1288
Current ANSI codepage : 1251
Plugins : 
    ComparePlugin (2.0.2)
    EmmetNPP (1.0.2)
    HexEditor (0.9.12)
    mimeTools (3.1)
    NppConverter (4.6)
    NppExport (0.4)
    NPPJSONViewer (1.40)
    PythonScript (2)
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by Trubisky**
Fixes #15886 and also #5089 by providing an auto-recover attempt for missing backup files relative to the CWD. The patch causes NppIO.cp to attempt to find and load a backup file from currentWorkingDirectioy/backup/[file name] if the session it's trying to load has a backup file path as part of it, but no backup file exists at the expected file path.

From my testing, seems to work as expected without modifying or breaking the functionality of anything else.

### CODE REVIEW

**Comment by Trubisky** [comment]
My original test commit pushed an older version of the NppIO file, and the second commit resolves that to be newest and include added functionality.

**Comment by alankilborn** [comment]
@Trubisky 

> and also 5089

??

File: PowerEditor/src/NppIO.cpp [hunk 1]
```diff
@@ -2399,9 +2399,36 @@ bool Notepad_plus::loadSession(Session & session, bool isSnapshotMode, const wch
...
 		else
 		{
-			BufferID foundBufID = MainFileManager.getBufferFromName(pFn);
-			if (foundBufID == BUFFER_INVALID)
-				lastOpened = nppGUI._keepSessionAbsentFileEntries ? MainFileManager.newPlaceholderDocument(pFn, MAIN_VIEW, userCreatedSessionName) : BUFFER_INVALID;
+			bool recoveredBackup = false;
+			//If it's in snapshot mode, there's a backupFilePath present, but the existing path doesn't exist
+			if (isSnapshotMode && !session._mainViewFiles[i]._backupFilePath.empty())
+			{
+				//Jusko was here, December 2024 <3
```
Comment (alankilborn) [inline PowerEditor/src/NppIO.cpp:2406]: I don't think such a comment is going to be acceptable.

Comment (xomx) [reply PowerEditor/src/NppIO.cpp:2406]: No offense, but this is just so similar to a doggy does when it marks its spots outside...  :-)

Comment (donho) [reply PowerEditor/src/NppIO.cpp:2406]: @xomx Could you provide the link please?

Comment (xomx) [reply PowerEditor/src/NppIO.cpp:2406]: @donho 
I meaned the same thing (`//Jusko was here, December 2024 <3`) as @alankilborn pointed out.
It just seems ridiculously similar to what dogs do outside when they want to let other dogs know they have been there... :-)

Anyway, due to the extensive reformatting (right one, though) in this PR, it took me a while to find a place with any relevant changes to the code.

**Comment by rust2** [comment]
Does this work if 

> @Trubisky
> 
> > and also 5089
> 
> ??

see #5098

File: PowerEditor/src/NppIO.cpp (hunk 1 above)
Comment (Trubisky) [reply PowerEditor/src/NppIO.cpp:2406]: No offense taken. Is this PR okay as is, or would the two of you rather I revise it without the commentary and with only relevant code changes (and not the reformatting)?

**Comment by Trubisky** [comment]
Yeah, 5098. Also, #1587

File: PowerEditor/src/NppIO.cpp (hunk 1 above)
Comment (donho) [reply PowerEditor/src/NppIO.cpp:2406]: > //Jusko was here, December 2024 <3

If I allow it, Notepad++ source code will be a graffiti wall.

**Comment by donho** [comment]
> //Jusko was here, December 2024 <3

If I allowed it, Notepad++ source code would be a graffiti wall.
PR rejected.

**Comment by alankilborn** [comment]
>//Jusko was here, December 2024 <3

I'm not even sure how somebody in the act of typing that would think it was a good idea.  Wowza.

**Pull request closed without merging**